torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NLCD dataset."""
+
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.colors import ListedColormap
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NLCD(RasterDataset):
+    """National Land Cover Database (NLCD) dataset.
+
+    The `NLCD dataset
+    <https://www.usgs.gov/centers/eros/science/national-land-cover-database>`_
+    is a land cover product that covers the United States and Puerto Rico. The current
+    implementation supports maps for the continental United States only. The product is
+    a joint effort between the United States Geological Survey
+    (`USGS <https://www.usgs.gov/>`_) and the Multi-Resolution Land Characteristics
+    Consortium (`MRLC <https://www.mrlc.gov/>`_) which released the first product
+    in 2001 with new updates every five years since then.
+
+    The dataset contains the following 15 classes:
+
+    #. Open Water
```

Comment:
It looks like the mask returned by `__getitem__` will start with 0, 1, 2, whereas this list will start with 1, 2, 3. Let's just manually label it 0, 1, 2 so that it matches and there is no confusion.

Reply:
Or is 0 a fill class? Maybe we should mention that there are actually 17 classes (0 being fill).

Reply:
Yes 0 is a background class, I believe, will adapt the docs.